### PR TITLE
New footer layout

### PR DESCRIPTION
**Pull request by barbaraperic**
## Visuals
Screenshot 2024-07-17 at 11 12 09
Screenshot 2024-07-17 at 11 12 02
Screenshot 2024-07-17 at 11 11 55

### CODE REVIEW

**Comment by CharlyMartin** [comment]
Shouldn't the footer button read `Subscribe to The Upload`?

Instead of `Subscribe to Newsletter` currently.

---

CleanShot 2024-07-17 at 17 08 54

**Comment by CharlyMartin** [comment]
I added a few CRs to (try to) simplify and harmonize the code, but overall it looks good :)

**Review by CharlyMartin — APPROVED** [review]
I added a few CRs to (try to) simplify and harmonize the code, but overall it looks good :)

**Comment by mirhamasala** [comment]
>Shouldn't the footer button read Subscribe to The Upload?

@CharlyMartin Where are you seeing that? [>>](https://www.figma.com/design/0pBfQs5tKI6bx5ypPIfVRA/FIL.org-V4-Prototypes?node-id=168-6670&t=9m9pLfwHdkbjNwsT-4)

File: src/app/_components/MobileNavigation.tsx [hunk 1]
```diff
@@ -73,23 +79,17 @@ type NestedMenuItemProps = {
 
 type NestedMenuProps = {
   title: string
-  items: NestedMenuItemProps[]
+  items: Array<NestedMenuItemProps>
```
Comment (mirhamasala) [inline src/app/_components/MobileNavigation.tsx:82]: @CharlyMartin ~ We recently went in and changed all of this to `[]` - Barbara mentioned she prefers the simplicity of it - And I think I agree. Any reason for the `Array` use?

File: src/app/_components/MobileNavigation.tsx [hunk 2]
```diff
@@ -73,23 +79,17 @@ type NestedMenuItemProps = {
...
-          <LinkItem
-            key={item.path}
-            nested
-            label={item.label}
-            path={item.path}
-            setOpen={setOpen}
-          />
+      <ul className="items-center space-y-6 border-l">
+        {items.map((item, i) => (
+          <LinkItem key={i} nested {...item} setOpen={setOpen} />
```
Comment (mirhamasala) [inline src/app/_components/MobileNavigation.tsx:92]: @CharlyMartin - Another one - I think O prefer seeing what's passed down..

**Review by mirhamasala — APPROVED** [review]
@CharlyMartin - Thanks so much for the extra review - This was cool doing it all together. Let's get this merged @barbaraperic (We can go without the Career link for now)! ✨

**Comment by CharlyMartin** [comment]
> @CharlyMartin Where are you seeing that? [>>](https://www.figma.com/design/0pBfQs5tKI6bx5ypPIfVRA/FIL.org-V4-Prototypes?node-id=168-6670&t=9m9pLfwHdkbjNwsT-4)

@mirhamasala The FIl+ mockup on Figma has a footer with `Subscribe to The Upload`, other pages don't. So it must be a mistake :)

File: src/app/_components/MobileNavigation.tsx (hunk 1 above)
Comment (CharlyMartin) [reply src/app/_components/MobileNavigation.tsx]: Ah my bad, I didn't know.

I prefer `Array` now for the readability aspect. I read [this article](https://tkdodo.eu/blog/array-types-in-type-script) a while ago about it.

> We usually read from left to right, so the things that are more important should come first. We say: "this is an array of strings"

Happy with `[]` too!

File: src/app/_components/MobileNavigation.tsx (hunk 2 above)
Comment (CharlyMartin) [reply src/app/_components/MobileNavigation.tsx]: Not sure I have a strong preference for either, it just made sense in the moment :)

File: src/app/_components/MobileNavigation.tsx (hunk 1 above)
Comment (mirhamasala) [reply src/app/_components/MobileNavigation.tsx]: Thank you for sharing the article - Like the say - Let's implement at least https://typescript-eslint.io/rules/array-type/

Comment (mirhamasala) [reply src/app/_components/MobileNavigation.tsx]: And let's create a ticket to talk about the article - It makes sense to me to move on to your notation.

Comment (barbaraperic) [reply src/app/_components/MobileNavigation.tsx]: Happy to go with `Array`, my comment before was about using one format. Thanks for sharing @CharlyMartin!

**Pull request merged**